rtc_uenclave/src/enclaves/rtc_auth.rs

```diff
@@ -47,4 +48,49 @@ where
     pub fn geteid(&self) -> sgx_enclave_id_t {
         self.0.geteid()
     }
+
+    /// Issues an execution token using the provided payload
+    pub fn issue_execution_token(
+        &self,
+        payload: &[u8],
+        metadata: ExecReqMetadata,
+    ) -> Result<EncryptedMessage, EcallError<ExecTokenError>> {
+        ecalls::issue_execution_token(self.geteid(), payload, metadata)
+    }
+}
+
+pub mod ecalls {
+    use auth_sys::ffi;
+    use rtc_types::*;
+    use sgx_types::*;
+
+    pub fn issue_execution_token(
+        eid: sgx_enclave_id_t,
+        payload: &[u8],
+        metadata: ExecReqMetadata,
+    ) -> Result<EncryptedMessage, EcallError<ExecTokenError>> {
+        let mut out_token_buffer = vec![0u8; 416].into_boxed_slice();
```

Comment:
Worth commenting where the 416 constant comes from?

(Same for the use in the length check?)